service/worker/replicator/replicationTask.go

```diff
@@ -435,8 +452,19 @@ func (t *historyReplicationV2Task) HandleErr(err error) error {
 	stopwatch := t.metricsClient.StartTimer(metrics.HistoryRereplicationByHistoryReplicationScope, metrics.CadenceClientLatency)
 	defer stopwatch.Stop()
 
-	// TODO execute v2 resend logic here before calling execute again
-
+	if err := t.nDCHistoryResender.SendSingleWorkflowHistory(
+		retryErr.GetDomainId(),
+		retryErr.GetWorkflowId(),
+		retryErr.GetRunId(),
+		retryErr.StartEventId,
+		retryErr.StartEventVersion,
+		retryErr.EndEventId,
+		retryErr.EndEventVersion,
+	); err != nil {
+		t.logger.Error("error resend history", tag.Error(err))
+		// should return the replication error, not the resending error
+		return err
```

Comment:
the err here is the resend err, not the original replication err

Reply:
Goo catch. Updated to return the origin error